sprocs/assessment_instances_insert.sql

```diff
@@ -26,19 +29,50 @@ BEGIN
 
     -- ######################################################################
     -- determine the "number" of the new assessment instance
+    IF group_work THEN
+        SELECT
+            gu.group_id
+        INTO
+            tmp_group_id
+        FROM
+            group_users as gu
+            JOIN groups AS g ON (g.id = gu.group_id)
+            JOIN group_configs AS gc ON (gc.id = g.group_config_id)
+        WHERE 
+            gu.user_id = assessment_instances_insert.user_id
+            AND gc.assessment_id = assessment_instances_insert.assessment_id
+            AND gc.deleted_at IS NULL
+            AND g.deleted_at IS NULL;
+            IF NOT FOUND THEN
+                RAISE EXCEPTION 'no matched group_id with user_id: %', assessment_instances_insert.user_id;
+            END IF;
+    END IF;    
 
     IF assessment.multiple_instance THEN
         SELECT coalesce(max(ai.number), 0) + 1
         INTO number
         FROM assessment_instances AS ai
         WHERE
             ai.assessment_id = assessment_instances_insert.assessment_id
-            AND ai.user_id = assessment_instances_insert.user_id;
+            AND (CASE 
+                    WHEN group_work THEN ai.group_id = tmp_group_id 
+                    ELSE ai.user_id = assessment_instances_insert.user_id
+                 END);
     END IF;
+        -- if a.multiple_instance is FALSE then we use
```

Comment:
This comment should not be indented. Please also leave a blank line before the comment.